core/granite/commons/sysInfo.ts

```diff
@@ -69,3 +71,19 @@ export function getRecommendedModels(systemInfo: SystemInfo) {
     defaultEmbeddingsModel: "nomic-embed-text:latest",
   };
 }
+
+export function shouldRecommendLargeModel(systemInfo: SystemInfo): boolean {
+  const requirements = MODEL_REQUIREMENTS[DEFAULT_MODEL_GRANITE_LARGE.model];
+  if (!requirements) return false;
+
+  if (isHighEndApple(systemInfo.gpus))
+    return systemInfo.memory.totalMemory >= requirements.recommendedMemoryBytes;
+
+  if (!hasDiscreteGPU(systemInfo.gpus)) return false;
+
+  const maxVRAM = systemInfo.gpus.reduce((max, gpu) => {
+    return Math.max(max, (gpu.vram || 0) * GB);
+  }, 0);
+
+  return maxVRAM >= requirements.recommendedMemoryBytes;
```

Comment:
I don't think we should be using the same number for the VRAM threshold for a discrete GPU and the memory requirement for a Mac (or eventually other iGPU systems). 12GB of VRAM for a discrete GPU is quite reasonale for the :8b model. 12GB of RAM is not.

Reply:
okay will add a 6GB offset for other tasks on system when using system memory

Reply:
so truthfully the 8b model takes like 8gb of memory and we have it set like:

```
    minMemoryBytes: 12 * GB,
    recommendedMemoryBytes: 16 * GB,
```
I think we should just have 1 number there, make it represent the actual usage of the model in memory, then have two model agnostic "reserve" numbers for amounts of the allocation of system and graphics memory that we won't encroach on when we do model selection.